lib/ui/window/platform_isolate.cc

```diff
@@ -0,0 +1,57 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <iostream>
+#include <thread>
+
+#include "flutter/fml/macros.h"
+#include "flutter/fml/task_runner.h"
+#include "flutter/lib/ui/ui_dart_state.h"
+#include "flutter/lib/ui/window/platform_isolate.h"
+#include "third_party/tonic/converter/dart_converter.h"
+
+namespace flutter {
+
+void PlatformIsolateNativeApi::Spawn(Dart_Handle entry_point,
+                                     Dart_Handle isolate_ready_port,
+                                     Dart_Handle debug_name) {
+  UIDartState* current_state = UIDartState::Current();
+  FML_DCHECK(current_state != nullptr);
+  if (!current_state->IsRootIsolate()) {
+    // TODO(flutter/flutter#136314): Remove this restriction.
+    Dart_EnterScope();
+    Dart_ThrowException(tonic::ToDart(
+        "PlatformIsolates can only be spawned on the root isolate."));
+  }
+
+  Dart_Port isolate_ready_port_id;
+  Dart_SendPortGetId(isolate_ready_port, &isolate_ready_port_id);
+  const char* debug_name_cstr;
+  Dart_StringToCString(debug_name, &debug_name_cstr);
+
+  char* error = nullptr;
+  current_state->CreatePlatformIsolate(entry_point, isolate_ready_port_id,
+                                       debug_name_cstr, &error);
+  if (error) {
+    Dart_EnterScope();
+    Dart_Handle error_handle = tonic::ToDart<const char*>(error);
+    ::free(error);
+    Dart_ThrowException(error_handle);
+  }
+}
+
+bool PlatformIsolateNativeApi::IsRunningInPlatformThread() {
+  UIDartState* current_state = UIDartState::Current();
+  if (current_state == nullptr) {
+    return false;
```

Comment:
Can't this be
```
FML_CHECK(UIDartState::Current() != nullptr);
```

This is a ffi native call, hence from an isolate, hence it has the isolate state attached.

Reply:
> I guess? I don't really see a reason to do that though. It's easy to handle this case by returning false.

Reading this code indicates that the state can be `nullptr` - so a reader may think in what circumstances this can be true and may be confused. If this is (according to our understanding) not possible, we should assert it.

In general adding "conservative code" is a bad smell:
* it's an indication that we do not understand the code or the invariants that hold well enough
* it introduces dead code that makes code less readable
* the added code isn't exercised in tests (as it's dead)
* if the (what we think is) dead code is actually exercised, strange things may happen, because a) we don't think it should run b) we don't have tests that do run it